sympy/series/series.py

```diff
@@ -10,3 +13,147 @@ def series(expr, x=None, x0=0, n=6, dir="+"):
+
+    x : variable, in terms of which equation is given, and in terms of which, the
+    expansion will be provided.
+
+    a : point on x-axis in the neighbourhood of which, the inverse expansion is to be found.
+
+    n : no of terms required in the expansion; default value is 3 terms.
+
+    Examples
+    ========
+
+    >>> from sympy.series.series import lagrange_inversion_theorem
+    >>> from sympy import exp
+    >>> from sympy.abc import x
+
+    Let the equation whose inverse's expansion is to be found be y = f(x)
+    then the first argument to be supplied must be f(x).
+    Like here, the equation whose inverse's expansion to be found is y = (x**2)*exp(x)
+
+    >>> eq = (x**2)*exp(x)
+
+    Here the number of terms required in the expansion is provided.
+
+    >>> lagrange_inversion_theorem(eq, x, 1, 4)
+    2*(x**2*exp(x) - E)**3*exp(-3)/27 - 7*(x**2*exp(x) - E)**2*exp(-2)/54 + (x**2*exp(x) - E)*exp(-1)/3 + 1
+
+    Here the number of terms required is, by default, 3.
+
+    >>> lagrange_inversion_theorem(eq, x, 1)
+    -7*(x**2*exp(x) - E)**2*exp(-2)/54 + (x**2*exp(x) - E)*exp(-1)/3 + 1
+
+    Here the point with respect to which the expansion is found is changed from x=1 to x=2.
+
+    >>> lagrange_inversion_theorem(eq, x, 2)
+    -7*(x**2*exp(x) - 4*exp(2))**2*exp(-4)/512 + (x**2*exp(x) - 4*exp(2))*exp(-2)/8 + 2
+
+    >>> eq = exp(x)
+    >>> lagrange_inversion_theorem(eq, x, 2)
+    -(exp(x) - exp(2))**2*exp(-4)/2 + (exp(x) - exp(2))*exp(-2) + 2
+
+    Here the number of terms and the point with respect to which the inverse expansion will be calculated are both
+    not provided explicitly.
+
+    >>> lagrange_inversion_theorem(eq, x)
+    -(exp(x) - 1)**2/2 + exp(x) - 1
+
+    Note
+    ====
+
+    There is a straightforward derivation using complex analysis and contour integration;
+    the complex formal power series version is a consequence of knowing the formula for polynomials,
+    so the theory of analytic functions may be applied. Actually, the machinery from analytic
+    function theory enters only in a formal way in this proof, in that what is really needed is
+    some property of the formal residue, and a more direct formal proof is available.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Lagrange_inversion_theorem
+    .. [2] http://mathworld.wolfram.com/LagrangeInversionTheorem.html
+    .. [3] http://www.cfm.brown.edu/people/dobrush/am33/Mathematica/lit.html
+    """
+
+    try:
+        not eq.free_symbols
+    except AttributeError:
+        raise ValueError("The function provided is not analytic i.e. f' = 0. Lagrange's inversion theorem requires "
+                         "that the function be analytic")
+
+    if len(eq.free_symbols) == 1:
+        if x not in eq.free_symbols:
+            raise ValueError("The function variable provided does not match the equation variable")
+
+    w = x
+    eq = eq.subs(x, w)
+    z = eq
```

Comment:
Do the lines above do anything? It looks to me as if they aren't needed.

Reply:
@oscarbenjamin  Everywhere I saw, the usage of the formula was in terms of `z` and `w`.  But everyone is comfortable in using `x` as a global variable for inputs. Hence the change as I didn't want anything to go wrong or difficult in terms of understanding the code by someone else or even by me while debugging.

Reply:
I think saying `w=x` and `z=eq` is fine if you want to use different variables names inside the function. The call to `subs` is just wasting CPU cycles though.

Reply:
the subs is done to replace `x` with `w` so that the problem can be solved (since everywhere I found the description of the formula in terms of `w` and `z`). If I wouldn't have used `subs` then how would I have used the equation? (Then later, the `w` is replaced back with `x`)

Reply:
Do you think this can be merged now?